docs/en/03_Upgrading/01_Upgrading_project.md

```diff
@@ -0,0 +1,1188 @@
+[2018-06-06 13:35:38] Applying ApiChangeWarningsRule to Coyote.php...
+modified:	Coyote.php
+@@ -68,7 +68,7 @@
+     {
+         // Getting a reference to Coyote's list of crazy ideas
+-        $manyManyRelation = $this->manyManyComponent('CrazyIdeas');
++        $manyManyRelation = $this->getSchema()->manyManyComponent('CrazyIdeas');
+         return $manyManyRelation;
+     }
+
+Warnings for Coyote.php:
+ - Coyote.php:20 SS_Cache: Using symfony/cache now (https://docs.silverstripe.org/en/4/changelogs/4.0.0#cache)
+ - Coyote.php:42 SilverStripe\Control\Director::setUrlParams(): Method removed
+ - Coyote.php:71 SilverStripe\ORM\DataObject->manyManyComponent(): DataObject->manyManyComponent() moved to DataObjectSchema. Access through getSchema(). You must manually add static::class as the first argument to manyManyComponent()
+Changes not saved; Run with --write to commit to disk
+```
+
+### Manually update deprecated API references
+
+SilverStripe 4 introduces many API changes. To update deprecated API references manually, you have to go through each one of your project files.
+Read the changelogs for [4.0.0](/changelogs/4.0.0/), [4.1.0](/changelogs/4.1.0/) and [4.2.0](/changelogs/4.2.0/)
+for a comprehensive overview.
+
+### Finalising the deprecated API update
+At this stage, your site should be using only SilverStripe 4 API logic.
+
+You still have some minor clean up tasks and configuration tweaks to apply, but you're almost done.
+
+This is a good point to commit your changes to your source control system before moving on to the next step.
+
+## Step 6 - Update your entry point {#step6}
+The location of SilverStripe's _entry file_ has changed. Your project and server environment will need
+to adjust the path to this file from `framework/main.php` to `index.php`.
+
+### Update your `index.php` file
+You can get a copy of the SilverStripe 4 `index.php` file at
+`vendor/silverstripe/recipe-core/public/index.php`.
+
+If you've created your own `index.php` or modified version of `main.php`,
+you'll need to reconcile those changes with the `index.php` file you got from `recipe-core`.
+Otherwise, just use the generic `index.php` file `recipe-core` provides.
+
+Copy your new `index.php` to your project's web root. Unlike SilverStripe 3, `index.php` must be present in your web root.
+
+### Update your server configuration
+If you're using a `.htaccess` file or `web.config` file to handle your server configuration, you can get the generic SilverStripe 4 version of those file from
+`vendor/silverstripe/recipe-core/public`.
+
+Just like `index.php`, if you've modified your server configuration file from the one that shipped with SilverStripe 3, you'll need to reconcile your changes into the version retrieve from `recipe-core`.
+
+[Refer to the installation instruction for your platform](/getting_started/installation/) if your server configuration is not managed via a `.htaccess` or `web.config` file.
```

Comment:
Can we also mention in this section the preference to set the web root to `public` rather than the root of the project? I remember someone (possibly Damian) saying that the .htaccess in the root folder was a sort of hacky fallback.

Reply:
Ok I see that's below - maybe add a link to that section?

Reply:
There's an assumption in the guide that you will read it sequentially (admittedly not a best practice when writing web content). 

If we start referencing other content that might be relevant further down or higher up, we'll just keep peppering same page links all over the place.